posydon/interpolation/profile_interpolation.py

```diff
@@ -0,0 +1,553 @@
+            profiles (array-like) : all N specified profiles, shape (N,200).
+        """
+        # open individual run as a DataFrame
+        df = pd.DataFrame(grid[ind]['final_profile1'])
+        df = df.sort_values(by="mass")
+        df = df.reset_index(drop=True)
+
+        # grab input values, final star 1 state, final star 1 mass
+        m1 = grid.initial_values["star_1_mass"][ind]
+        m2 = grid.initial_values["star_2_mass"][ind]
+        p = grid.initial_values["period_days"][ind]
+        state = grid.final_values["S1_state"][ind]  
+        final_mass = grid.final_values["star_1_mass"][ind]
+        scalars = {"m1":m1,
+                   "m2":m2,
+                   "p":p,
+                   "s1_state":state,
+                   "final_mass":final_mass}
+
+        # grab output vectors, interpolate to normalize
+        profiles=np.zeros([len(self.names),200])
+        for i,prof in enumerate(self.names):
+            f = interp1d(df['mass']/final_mass,df[prof],fill_value="extrapolate")
+            profile_new = f(np.linspace(0,1,200))
+            profiles[i] = profile_new
+
+        return scalars, profiles
+    
+    def save(self, filename):
+        """Save extracted profile data.
+        Args:
+            filename (str) : path/name of '.pkl' file where the data will be saved.
+        """
+        SAVE_ATTRS = self.__dict__
+        DONT_SAVE = []
+        myattrs = {key: SAVE_ATTRS[key]
+                   for key in SAVE_ATTRS if key not in DONT_SAVE}
+        with open(filename, 'wb') as f:
+            pickle.dump(myattrs, f)
+
+            
+class ProfileInterpolator:
+    
+    def __init__(self,load_interpolator=None):
+        """Interfaces with other classes, trains models and predicts profiles.
+        Args:
+            load_interpolator (str) : optional path/name of interpolator file to be loaded
+        """
+        
+        if load_interpolator != None:
+            self.load(load_interpolator)
+        
+    def load_profiles(self,filename):
+        """Load extracted profile data.
+        Args:
+            filename (str) : path/name of '.pkl' file to be loaded.
+        Returns:
+            self.profiles (array-like) : M profiles for each of N training binaries, shape (N,M,200).
+            self.scalars (array-like) : DataFrame containing initial and final conditions for N training binaries.
+            self.valid_profiles (array-like) : M profiles for each of N testing binaries, shape (N,M,200).
+            self.valid_scalars (array-like) : DataFrame containing initial and final conditions for N testing binaries.
+        """
+        with open(filename, 'rb') as f:
+            myattrs = pd.read_pickle(f)
+            for key in myattrs:
+                setattr(self, key, myattrs[key])
+        self.profiles = np.array(self.profiles)
+        self.valid_profiles = np.array(self.valid_profiles)
+        return self.profiles,self.scalars,self.valid_profiles,self.valid_scalars
```

Comment:
I'm not sure, whether it is a good idea to return here this 4 parts. Wouldn't is be better to return `self`, in this way you wouldn't discard other key (you may add in the future). And you won't need to create new objects on the fly.
But I see, where you called it you don't use the return value at all.